core/string/src/lib.rs

```diff
@@ -149,37 +150,83 @@ impl CodePoint {
     }
 }
 
-/// The raw representation of a [`JsString`] in the heap.
-#[repr(C)]
-struct RawJsString {
-    /// Contains the flags and Latin1/UTF-16 length.
-    ///
-    /// The latin1 flag is stored in the bottom bit.
-    flags_and_len: usize,
+/// Contains the flags and Latin1/UTF-16 length.
+///
+/// The latin1 flag is stored in the bottom bit.
```

Comment:
Nit: I would change the first line to a more "descriptive" style. Something like "A length with a tag on its bottom bit". The second line looks good.